.github/workflows/spread-docs.yaml

```diff
@@ -14,25 +14,70 @@ on:
     - cron: "0 0 * * 0" # Midnight UTC on Sundays
 
 jobs:
+  snap-build:
+    runs-on: [self-hosted, self-hosted-linux-amd64-noble-medium]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Build snap
+        uses: snapcore/action-build@v1
+        id: charmcraft
+      - name: Upload snap artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: snap
+          path: ${{ steps.charmcraft.outputs.snap }}
+
   spread-select:
-    runs-on: spread-installed
+    runs-on: [self-hosted, self-hosted-linux-amd64-noble-medium]
     outputs:
       systems: ${{ steps.select.outputs.systems }}
     steps:
+      - name: Checkout charmcraft
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          submodules: true
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.24
+      - name: Install spread
+        run: |
+          go install github.com/snapcore/spread/cmd/spread@latest
       - name: Find spread systems
         id: select
         run: |
-          echo systems=$(spread -list google:docs/ | cut -d: -f2 | sort | uniq | jq --raw-input . | jq --slurp --compact-output) | tee -a "${GITHUB_OUTPUT}"
+          echo systems=$($(go env GOPATH)/bin/spread -list ci:docs/ | jq --raw-input . | jq --slurp --compact-output) | tee -a "${GITHUB_OUTPUT}"
 
   docs-tests:
-    runs-on: spread-installed
-    needs: [spread-select]
+    runs-on: [self-hosted, self-hosted-linux-amd64-noble-large]
+    needs: [spread-select, snap-build]
+    timeout-minutes: 90
     strategy:
       fail-fast: false
       matrix:
         system: ${{ fromJSON(needs.spread-select.outputs.systems) }}
 
     steps:
+      - name: Fix network issues related to docker and lxd
+        run: |
+          sudo iptables -P FORWARD ACCEPT
+          sudo ip6tables -P FORWARD ACCEPT
+      - name: Checkout charmcraft
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          submodules: true
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.24
+      - name: Install spread
+        run: |
+          go install github.com/snapcore/spread/cmd/spread@latest
```

Comment:
Can we use the `spread` binary installed in the charmcraft snap instead? (Available as `charmcraft.spread`)